hpedockerplugin/file_manager.py

```diff
@@ -0,0 +1,486 @@
+                               u"Mountpoint": mount_dir,
+                               u"Devicename": share_path})
+        return response
+
+    def unmount_share(self, share_name, share, mount_id):
+        # Start of volume fencing
+        LOG.info('Unmounting share: %s' % share)
+        # share = {
+        #   'path_info': {
+        #     node_id1: {'mnt_id1': 'mnt_dir1', 'mnt_id2': 'mnt_dir2',...},
+        #     node_id2: {'mnt_id2': 'mnt_dir2', 'mnt_id3': 'mnt_dir3',...},
+        #   }
+        # }
+        path_info = share.get('path_info')
+        if path_info:
+            node_mnt_info = path_info.get(self._node_id)
+            if node_mnt_info:
+                mount_dir = node_mnt_info.get(mount_id)
+                if mount_dir:
+                    LOG.info('Unmounting share: %s...' % mount_dir)
+                    sh.umount(mount_dir)
+                    LOG.info('Removing dir: %s...' % mount_dir)
+                    sh.rm('-rf', mount_dir)
+                    LOG.info("Removing mount-id '%s' from meta-data" %
+                             mount_id)
+                    del node_mnt_info[mount_id]
+
+                # If this was the last mount of share share_name on
+                # this node, remove my_ip from client-ip list
+                if not node_mnt_info:
+                    del path_info[self._node_id]
+                    my_ip = netutils.get_my_ipv4()
+                    LOG.info("Remove %s from client IP list" % my_ip)
+                    client_ips = share['clientIPs']
+                    client_ips.remove(my_ip)
+                    self._hpeplugin_driver.remove_client_ip_for_share(
+                        share['id'], my_ip)
+                    # If this is the last node from where share is being
+                    # unmounted, remove the path_info from share metadata
+                    if not path_info:
+                        del share['path_info']
+                LOG.info('Share unmounted. Updating ETCD: %s' % share)
+                self._etcd.save_share(share)
+                LOG.info('Unmount DONE for share: %s, %s' %
+                         (share_name, mount_id))
+            else:
+                LOG.error("ERROR: Node mount information not found in ETCD")
+        else:
+            LOG.error("ERROR: Path info missing from ETCD")
+        response = json.dumps({u"Err": ''})
+        return response
+
+    def import_share(self, volname, existing_ref, backend='DEFAULT',
+                     manage_opts=None):
+        pass
+
+    @staticmethod
+    def _rollback(rollback_list):
+        for undo_action in reversed(rollback_list):
+            LOG.info(undo_action['msg'])
+            try:
+                undo_action['undo_func'](**undo_action['params'])
+            except Exception as ex:
+                # TODO: Implement retry logic
+                LOG.exception('Ignoring exception: %s' % ex)
+                pass
+
+    def _decrypt(self, encrypted, passphrase):
```

Comment:
function definitions `_decrypt(), _decrypt_password(), key_check()` can move a common class and the `FileManager` and `VolumeManager` can inherit from this class. Reason being these functions are duplicated between both the managers and developers might later need to fix some issues at 2 different places.

Reply:
I agree that the common implementation should not be duplicated. Moving out this to a separate class PasswordDecryptor